x-pack/plugins/observability/server/plugin.ts

```diff
@@ -14,23 +20,76 @@ import {
   AnnotationsAPI,
 } from './lib/annotations/bootstrap_annotations';
 import type { RuleRegistryPluginSetupContract } from '../../rule_registry/server';
+import { PluginSetupContract as FeaturesSetup } from '../../features/server';
 import { uiSettings } from './ui_settings';
 import { registerRoutes } from './routes/register_routes';
 import { getGlobalObservabilityServerRouteRepository } from './routes/get_global_observability_server_route_repository';
 
 export type ObservabilityPluginSetup = ReturnType<ObservabilityPlugin['setup']>;
 
+interface PluginSetup {
+  features: FeaturesSetup;
+  ruleRegistry: RuleRegistryPluginSetupContract;
+}
+
+const OBS_CASES_ID = 'observabilityCases';
+const OBSERVABILITY = 'observability';
+
 export class ObservabilityPlugin implements Plugin<ObservabilityPluginSetup> {
   constructor(private readonly initContext: PluginInitializerContext) {
     this.initContext = initContext;
   }
 
-  public setup(
-    core: CoreSetup,
-    plugins: {
-      ruleRegistry: RuleRegistryPluginSetupContract;
-    }
-  ) {
+  public setup(core: CoreSetup, plugins: PluginSetup) {
+    plugins.features.registerKibanaFeature({
+      id: OBS_CASES_ID,
+      name: i18n.translate('xpack.observability.featureRegistry.linkObservabilityTitle', {
+        defaultMessage: 'Cases',
+      }),
+      order: 1100,
+      category: DEFAULT_APP_CATEGORIES.observability,
+      app: [OBS_CASES_ID, 'kibana'],
```

Comment:
The value of `OBS_CASES_ID` does not match the name of the application that you register on the client-side. This means that users granted access to your Cases feature won't actually be able to use the Cases app. It just doesn't appear in the UI for them:

<img width="319" alt="image" src="https://user-images.githubusercontent.com/3493255/121202439-491e0400-c843-11eb-8cda-078c25d7820b.png">


This should match the id of the app(s) you intend to grant access to. I presume this is `observability-cases`?

Reply:
ahhh we were wondering why that was happing! Thanks @legrego ! @stephmilovic we'll want to switch `OBS_CASES_ID` to `CASES_APP_ID`